dags/utils/slack_operator.py

```diff
@@ -0,0 +1,81 @@
+from airflow.hooks.base_hook import BaseHook
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+
+# This is the Conn Id that we set when creating the connection in the Airflow dashboard
+# in Admin > Connections.
+SLACK_CONN_ID = "slack"
+
+
+def task_fail_slack_alert_critical(context):
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+    slack_msg = """
+            <!channel> :red_circle: Critical Failure
+            *Task*: {task}  
+            *Dag*: {dag} 
+            *Execution Time*: {exec_date}  
+            *Log Url*: {log_url} 
+            """.format(
+        task=context.get("task_instance").task_id,
+        dag=context.get("task_instance").dag_id,
+        ti=context.get("task_instance"),
+        exec_date=context.get("execution_date"),
+        log_url=context.get("task_instance").log_url,
+    )
+    failed_alert = SlackWebhookOperator(
+        task_id="slack_critical_failure",
+        http_conn_id="slack",
+        webhook_token=slack_webhook_token,
+        message=slack_msg,
+        username="airflow",
+    )
+    return failed_alert.execute(context=context)
+
+
+def task_fail_slack_alert(context):
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+    slack_msg = """
+            :red_circle: Task Failed. 
+            *Task*: {task}  
+            *Dag*: {dag} 
+            *Execution Time*: {exec_date}  
+            *Log Url*: {log_url} 
+            """.format(
+        task=context.get("task_instance").task_id,
+        dag=context.get("task_instance").dag_id,
+        ti=context.get("task_instance"),
+        exec_date=context.get("execution_date"),
+        log_url=context.get("task_instance").log_url,
+    )
+    failed_alert = SlackWebhookOperator(
+        task_id="slack_failure",
+        http_conn_id="slack",
+        webhook_token=slack_webhook_token,
+        message=slack_msg,
+        username="airflow",
+    )
```

Comment:
it would be really nice to get the actual python exception text in the slack message. i did a bit of searching (and recalling from memory) and i don't think there's a way to pull this out of the task `context`, and it would involve some heavy lifting. all that to say this looks great and replicates Airflow v1 perfectly ✨